### PR TITLE
CORE-8772 Logging when transaction is not persisted

### PR DESCRIPTION
**Pull request by mbrkic-r3**
- Logging when ledger repository inserts are ignored due to existing rows
- Refactored Consensual ledger code to match UTXO implementation

### CODE REVIEW

File: components/ledger/ledger-persistence/src/integrationTest/kotlin/net/corda/ledger/persistence/consensual/tests/ConsensualLedgerMessageProcessorTests.kt [hunk 1]
```diff
@@ -170,7 +170,7 @@ class ConsensualLedgerMessageProcessorTests {
         request: Any,
         externalEventContext: ExternalEventContext = EXTERNAL_EVENT_CONTEXT
     ): LedgerPersistenceRequest {
-        logger.info("Consensual ledger persistence request: {} {}", request.javaClass.simpleName, request)
+        logger.debug("Consensual ledger persistence request: {} {}", request.javaClass.simpleName, request)
```
Comment (lankydan) [inline components/ledger/ledger-persistence/src/integrationTest/kotlin/net/corda/ledger/persistence/consensual/tests/ConsensualLedgerMessageProcessorTests.kt]: Also change this to `debug { .. }`

File: components/ledger/ledger-persistence/src/main/kotlin/net/corda/ledger/persistence/consensual/ConsensualRepository.kt [hunk 2]
```diff
@@ -0,0 +1,76 @@
+package net.corda.ledger.persistence.consensual
+
+import net.corda.ledger.common.data.transaction.SignedTransactionContainer
+import net.corda.ledger.common.data.transaction.TransactionStatus
+import net.corda.v5.application.crypto.DigitalSignatureAndMetadata
+import net.corda.v5.ledger.common.transaction.CordaPackageSummary
+import java.time.Instant
+import javax.persistence.EntityManager
+
+interface ConsensualRepository {
```
Comment (lankydan) [inline components/ledger/ledger-persistence/src/main/kotlin/net/corda/ledger/persistence/consensual/ConsensualRepository.kt:10]: Can we get the git history to know this was a rename?

File: components/ledger/ledger-persistence/src/main/kotlin/net/corda/ledger/persistence/consensual/impl/ConsensualRepositoryImpl.kt [hunk 3]
```diff
@@ -28,25 +31,25 @@ import javax.persistence.Tuple
  * only the [UsedByPersistence] marker interface.
  */
 @Component(
-    service = [ ConsensualLedgerRepository::class, UsedByPersistence::class ],
+    service = [ ConsensualRepositoryImpl::class, UsedByPersistence::class ],
```
Comment (lankydan) [inline components/ledger/ledger-persistence/src/main/kotlin/net/corda/ledger/persistence/consensual/impl/ConsensualRepositoryImpl.kt]: This should be the interface.

File: components/ledger/ledger-persistence/src/main/kotlin/net/corda/ledger/persistence/utxo/impl/UtxoRepositoryImpl.kt [hunk 4]
```diff
@@ -303,6 +294,7 @@ class UtxoRepositoryImpl(
             .setParameter("isRefInput", isRefInput)
             .setParameter("createdAt", timestamp)
             .executeUpdate()
+            .logResult("transaction relevancy [$transactionId, $groupIndex, $leafIndex]")
```
Comment (vlajos) [inline components/ledger/ledger-persistence/src/main/kotlin/net/corda/ledger/persistence/utxo/impl/UtxoRepositoryImpl.kt]: Is this relevancy?

File: components/ledger/ledger-persistence/src/main/kotlin/net/corda/ledger/persistence/consensual/impl/ConsensualRepositoryImpl.kt [hunk 5]
```diff
@@ -231,27 +220,18 @@ class ConsensualLedgerRepository @Activate constructor(
...
-            """
-            SELECT file_checksum
-            FROM {h-schema}consensual_transaction_cpk
-            WHERE file_checksum in (:fileChecksums)""",
-            Tuple::class.java
-        )
-            .setParameter("fileChecksums", cpkMetadata.map { it.fileChecksum })
-            .resultListAsTuples()
-            .mapTo(HashSet()) { r -> r.get(0) as String }
+    private fun Int.logResult(entity: String): Int {
```
Comment (vlajos) [inline components/ledger/ledger-persistence/src/main/kotlin/net/corda/ledger/persistence/consensual/impl/ConsensualRepositoryImpl.kt:230]: `persistTransactionCpk` could call this logResult too.

File: components/ledger/ledger-persistence/src/integrationTest/kotlin/net/corda/ledger/persistence/consensual/tests/ConsensualLedgerMessageProcessorTests.kt (hunk 1 above)
Comment (mbrkic-r3) [reply components/ledger/ledger-persistence/src/integrationTest/kotlin/net/corda/ledger/persistence/consensual/tests/ConsensualLedgerMessageProcessorTests.kt]: fixed

File: components/ledger/ledger-persistence/src/main/kotlin/net/corda/ledger/persistence/consensual/ConsensualRepository.kt (hunk 2 above)
Comment (mbrkic-r3) [reply components/ledger/ledger-persistence/src/main/kotlin/net/corda/ledger/persistence/consensual/ConsensualRepository.kt:10]: Interface didn't exist before

File: components/ledger/ledger-persistence/src/main/kotlin/net/corda/ledger/persistence/consensual/impl/ConsensualRepositoryImpl.kt (hunk 3 above)
Comment (mbrkic-r3) [reply components/ledger/ledger-persistence/src/main/kotlin/net/corda/ledger/persistence/consensual/impl/ConsensualRepositoryImpl.kt]: fixed

File: components/ledger/ledger-persistence/src/main/kotlin/net/corda/ledger/persistence/utxo/impl/UtxoRepositoryImpl.kt (hunk 4 above)
Comment (mbrkic-r3) [reply components/ledger/ledger-persistence/src/main/kotlin/net/corda/ledger/persistence/utxo/impl/UtxoRepositoryImpl.kt]: fixed

File: components/ledger/ledger-persistence/src/main/kotlin/net/corda/ledger/persistence/consensual/impl/ConsensualRepositoryImpl.kt (hunk 5 above)
Comment (mbrkic-r3) [reply components/ledger/ledger-persistence/src/main/kotlin/net/corda/ledger/persistence/consensual/impl/ConsensualRepositoryImpl.kt:230]: fixed

**Review by vlajos — APPROVED** [review]

**Pull request merged**